test/ttbar/TTbarEventAnalysis.cc

```diff
@@ -127,479 +230,826 @@ void TTbarEventAnalysis::prepareOutput(TString outFile)
+    tree->SetBranchAddress("ttbar_lm"   ,   ev.ttbar_lm);
+    tree->SetBranchAddress("ttbar_lid"  ,   ev.ttbar_lid);
+    tree->SetBranchAddress("ttbar_lgid" ,   ev.ttbar_lgid);
+    tree->SetBranchAddress("ttbar_lch"  ,   ev.ttbar_lch);
+    tree->SetBranchAddress("ttbar_metpt",  &ev.ttbar_metpt);
+    tree->SetBranchAddress("ttbar_metphi", &ev.ttbar_metphi);
+    tree->SetBranchAddress("ttbar_rho",    &ev.ttbar_rho);
+    tree->SetBranchAddress("ttbar_nw",     &ev.ttbar_nw);
+    tree->SetBranchAddress("ttbar_w",      ev.ttbar_w);
+    tree->SetBranchAddress("nJet",            &ev.nJet);
+    tree->SetBranchAddress("Jet_pt",          ev.Jet_pt);
+    tree->SetBranchAddress("Jet_genpt",       ev.Jet_genpt);
+    tree->SetBranchAddress("Jet_area",        ev.Jet_area);
+    tree->SetBranchAddress("Jet_jes",         ev.Jet_jes);
+    tree->SetBranchAddress("Jet_eta",         ev.Jet_eta);
+    tree->SetBranchAddress("Jet_phi",         ev.Jet_phi);
+    tree->SetBranchAddress("Jet_mass",        ev.Jet_mass);
+    tree->SetBranchAddress("Jet_Svx",         ev.Jet_Svx);
+    tree->SetBranchAddress("Jet_CombIVF",     ev.Jet_CombIVF);
+    tree->SetBranchAddress("Jet_DeepCSVBDisc",ev.Jet_DeepCSVBDisc);
+    tree->SetBranchAddress("Jet_DeepFlavourBDisc",ev.Jet_DeepFlavourBDisc);
+    tree->SetBranchAddress("Jet_Proba",       ev.Jet_Proba);
+    tree->SetBranchAddress("Jet_Ip2P",        ev.Jet_Ip2P);
+    tree->SetBranchAddress("Jet_nseltracks",  ev.Jet_nseltracks);
+    tree->SetBranchAddress("Jet_flavour",     ev.Jet_flavour);
+
+    int nSkipped=0;
+    int n1=0;
+    int n2=0;
+    int n3=0;
+    for(Int_t i=0; i<nentries; i++){
+        tree->GetEntry(i);
+       
+        //if(isData){
+        //    if(ev.Run>304671){ //EF
+        //        nSkipped++;
+        //        continue;
+        //    }
+        //}
+    
+        //progress bar
+        if(i%100==0) std::cout << "\r[ " << int(100.*i/nentries) << "/100 ] to completion" << std::flush;
+ 
+        //generator level weights
+        Float_t genWgt=ev.ttbar_nw==0 ? 1.0 : ev.ttbar_w[0];
+        //std::cout<<"go go 001"<<i<<std::endl;       
+        if(readTTJetsGenWeights_ && ev.ttbar_nw>17) {
+            // Weight * [sample xsec from json / sum(weights for given systematic)] / [sample xsec from json / sum(weights for nominal)]
+            // = Weight * sum(weights nominal) / sum(weights for given systematic)
+            // i.e. renormalise to same number of events as nominal.
+            // N.B. for GetBinContent(X) 'X' must be weight index + 1 due to way histogram is filled (0th bin always underflow).
+            systWeight["qcdScaleLo"]=ev.ttbar_w[9]*(xsecWgt->GetBinContent(10)/xsecWgt->GetBinContent(1));
+            systWeight["qcdScaleHi"]=ev.ttbar_w[5]*(xsecWgt->GetBinContent(6)/xsecWgt->GetBinContent(1));
+            systWeight["hdampLo"]=ev.ttbar_w[ev.ttbar_nw-29]*(xsecWgt->GetBinContent(ev.ttbar_nw-29+1)/xsecWgt->GetBinContent(1));
```

Comment:
@Wilsker These weights don't seem correct.  From data/mc it is clear mc normalization is much too large.  Is there another factor?